### PR TITLE
[CIRC-427] Expand check-out endpoint error response with loan policy info

### PR DESCRIPTION
**Pull request by dmytrotur**
## Purpose
Add loan policy id and name to error response when cannot check-out item due to not loanable policy. 
Resolves: [CIRC-427](https://issues.folio.org/browse/CIRC-427)
Required for UI story: [UICHKOUT-534](https://issues.folio.org/browse/UICHKOUT-534)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.

### CODE REVIEW

File: src/test/java/api/loans/CheckOutByBarcodeTests.java [hunk 1]
```diff
@@ -942,4 +943,30 @@ public void checkOutDoesNotFailWhenCirculationRulesReferenceInvalidNoticePolicyI
...
+    throws InterruptedException,
+    MalformedURLException,
+    TimeoutException,
+    ExecutionException {
+
+    IndividualResource notLoanablePolicy = loanPoliciesFixture.create(
+      new LoanPolicyBuilder()
+        .withName("Not Loanable Policy")
+        .withLoanable(false));
+    useLoanPolicyAsFallback(
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/CheckOutByBarcodeTests.java]: Personally, I might find it easier to read if these statements were separated by a blank line

File: src/test/java/api/loans/CheckOutByBarcodeTests.java [hunk 2]
```diff
@@ -942,4 +943,30 @@ public void checkOutDoesNotFailWhenCirculationRulesReferenceInvalidNoticePolicyI
 
     assertThat(smallAngryPlanet, hasItemStatus(CHECKED_OUT));
   }
+
+  @Test
+  public void cannotCheckOutWhenItemIsNotLoanable()
```
Comment (marcjohnson-kint) [inline src/test/java/api/loans/CheckOutByBarcodeTests.java]: Thank you for adding this test (I'm a little surprised it didn't already exist)

File: src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java [hunk 3]
```diff
@@ -41,10 +44,17 @@
...
   private Result<LoanAndRelatedRecords> refuseWhenItemIsNotLoanable(LoanAndRelatedRecords relatedRecords) {
     final Loan loan = relatedRecords.getLoan();
+    final LoanPolicy loanPolicy = loan.getLoanPolicy();
 
-    if (loan.getLoanPolicy().isNotLoanable()) {
+    if (loanPolicy.isNotLoanable()) {
       String itemBarcode = loan.getItem().getBarcode();
-      return failed(singleValidationError("Item is not loanable", ITEM_BARCODE, itemBarcode));
+
+      Map<String, String> parameters = new HashMap<>();
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java]: I might be tempted to extract this into a method like `loanPolicyValidationError(loanPolicy, message, additionalParameters)`

And then replace usage of `LoanPolicy.errorForPolicy` with this.

Comment (dmytrotur) [reply src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java]: Makes sense for me. Do you think `LoanPolicy` class is an appropriate place for this?

File: src/test/java/api/loans/CheckOutByBarcodeTests.java (hunk 1 above)
Comment (dmytrotur) [reply src/test/java/api/loans/CheckOutByBarcodeTests.java]: Agree, added a blank line.

File: src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java (hunk 3 above)
Comment (marcjohnson-kint) [reply src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java]: Probably at this point, I'd prefer it wasn't but that is likely a bigger refactoring

Comment (dmytrotur) [reply src/main/java/org/folio/circulation/resources/RegularCheckOutStrategy.java]: Ok, so I'll leave it in `LoanPolicy` for now.

**Pull request merged**